# Copyright Contributors to the OpenVDB Project SPDX-License-Identifier: MPL-2.0

  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.
----
diff -urp nanovdb.orig/CMakeLists.txt nanovdb/CMakeLists.txt
--- nanovdb.orig/CMakeLists.txt	2020-09-24 20:19:30.000000000 -0700
+++ nanovdb/CMakeLists.txt	2021-03-30 20:10:41.707086955 -0700
@@ -77,13 +77,10 @@ endif()
 
 # -----------------------------------------------------------------------
 if(NANOVDB_BUILD_UNITTESTS OR NANOVDB_BUILD_BENCHMARK)
-  if(NANOVDB_ALLOW_FETCHCONTENT)
-    message(STATUS "Downloading GTest...")
+  if(NANOVDB_USE_INTERNAL_GTEST)
+    set(googletest_SOURCE_DIR ${EGOOGLETEST_SOURCE_DIR})
+    set(googletest_BINARY_DIR ${EGOOGLETEST_BINARY_DIR})
     include(FetchContent)
-    FetchContent_Declare(
-      googletest
-      GIT_REPOSITORY https://github.com/google/googletest.git
-      GIT_TAG release-1.10.0)
     set(INSTALL_GTEST
         OFF
         CACHE BOOL "" FORCE)
@@ -93,9 +90,7 @@ if(NANOVDB_BUILD_UNITTESTS OR NANOVDB_BU
     set(gtest_force_shared_crt
         ON
         CACHE BOOL "" FORCE)
-    FetchContent_GetProperties(googletest)
     if(NOT googletest_POPULATED)
-      FetchContent_Populate(googletest)
       add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
     endif()
     set(NANOVDB_GTEST "gtest" "gtest_main")
diff -urp nanovdb.orig/cmd/CMakeLists.txt nanovdb/cmd/CMakeLists.txt
--- nanovdb.orig/cmd/CMakeLists.txt	2020-09-24 20:19:30.000000000 -0700
+++ nanovdb/cmd/CMakeLists.txt	2021-03-30 20:07:34.379622448 -0700
@@ -25,17 +25,12 @@ if(NANOVDB_BUILD_INTERACTIVE_RENDERER)
     if(glfw3_FOUND)
       set(NANOVDB_GLFW_LINK_LIBRARY "glfw")
       set(NANOVDB_USE_GLFW_FLAG "NANOVDB_USE_GLFW")
-    elseif(NANOVDB_ALLOW_FETCHCONTENT)
+    elseif(NANOVDB_USE_INTERNAL_GLFW)
       if(NOT glfw_POPULATED)
-        message(STATUS "Downloading glfw...")
-        FetchContent_Declare(
-          glfw
-          GIT_REPOSITORY https://github.com/glfw/glfw
-          GIT_TAG 3.3)
-
-        FetchContent_GetProperties(glfw)
         if(NOT glfw_POPULATED)
-          FetchContent_Populate(glfw)
+          set(glfw_SOURCE_DIR ${EGLFW_SOURCE_DIR})
+          set(glfw_BINARY_DIR ${EGLFW_BINARY_DIR})
+
           set(GLFW_BUILD_EXAMPLES
               OFF
               CACHE INTERNAL "Build the GLFW example programs")
@@ -57,18 +52,10 @@ if(NANOVDB_BUILD_INTERACTIVE_RENDERER)
 
     # -----------------------------
     if(NANOVDB_USE_NATIVEFILEDIALOG)
-      if(NANOVDB_ALLOW_FETCHCONTENT)
+      if(NANOVDB_USE_NATIVEFILEDIALOG)
         if(NOT nfd_POPULATED)
-          message(STATUS "Downloading nfd...")
-
-          FetchContent_Declare(
-            nfd
-            GIT_REPOSITORY https://github.com/mlabbe/nativefiledialog.git
-            GIT_TAG master)
-
-          FetchContent_GetProperties(nfd)
           if(NOT nfd_POPULATED)
-            FetchContent_Populate(nfd)
+            set(nfd_SOURCE_DIR ${ENFD_SOURCE_DIR})
 
             if(APPLE)
               list(APPEND NFD_SOURCE_FILES ${nfd_SOURCE_DIR}/src/nfd_cocoa.m
@@ -113,17 +100,11 @@ if(NANOVDB_BUILD_INTERACTIVE_RENDERER)
   endif(NOT USE_EMSCRIPTEN_TOOLCHAIN)
 
   # -----------------------------
-  if(NANOVDB_ALLOW_FETCHCONTENT)
+  if(NANOVDB_USE_IMGUI)
     if(NOT imgui_POPULATED)
-      message(STATUS "Downloading imgui...")
-      FetchContent_Declare(
-        imgui
-        GIT_REPOSITORY https://github.com/ocornut/imgui.git
-        GIT_TAG master)
-
-      FetchContent_GetProperties(imgui)
       if(NOT imgui_POPULATED)
-        FetchContent_Populate(imgui)
+        set(imgui_SOURCE_DIR ${EIMGUI_SOURCE_DIR})
+        message(STATUS "${imgui_SOURCE_DIR}")
 
         set(NANOVDB_USE_IMGUI_FLAG "NANOVDB_USE_IMGUI")
 
